File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = s.length(), longest = 0;

        map<char, int> mp;
        int l = 0;

        for(int r = 0; r < n; r++){
            if(mp.find(s[r]) != mp.end()){
                l = max(l, mp[s[r]]+1);
            }
            mp[s[r]] = r;
            longest = max(longest, r-l+1);
        }
        return longest;
    }
};

File: Data Structures & Algorithms/eating-bananas/submission-0.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int len = piles.length;
        int low = 1, high = Arrays.stream(piles).max().getAsInt();

        while(low <= high){
            int mid = low + (high-low)/2;
            int res=0;
            for(int i = 0; i < len; i++){
                res += (int)Math.ceil((double)piles[i] / mid);
            }
            if(res > h)
                low = mid+1;
            else
                high = mid-1;
        }
        return low;
    }
}
